extensions/web/09_file.js

```diff
@@ -189,79 +190,88 @@
      * @param {string} [contentType]
      * @returns {Blob}
      */
-    slice(start, end, contentType) {
+    slice(start = 0, end = this.size, contentType = '') {
       webidl.assertBranded(this, Blob);
       const prefix = "Failed to execute 'slice' on 'Blob'";
-      if (start !== undefined) {
-        start = webidl.converters["long long"](start, {
-          clamp: true,
-          context: "Argument 1",
-          prefix,
-        });
-      }
-      if (end !== undefined) {
-        end = webidl.converters["long long"](end, {
-          clamp: true,
-          context: "Argument 2",
-          prefix,
-        });
-      }
-      if (contentType !== undefined) {
-        contentType = webidl.converters["DOMString"](contentType, {
-          context: "Argument 3",
-          prefix,
-        });
-      }
+      start = webidl.converters["long long"](start, {
+        clamp: true,
+        context: "Argument 1",
+        prefix,
+      });
+      end = webidl.converters["long long"](end, {
+        clamp: true,
+        context: "Argument 2",
+        prefix,
+      });
+      contentType = webidl.converters["DOMString"](contentType, {
+        context: "Argument 3",
+        prefix,
+      });
 
-      // deno-lint-ignore no-this-alias
-      const O = this;
-      /** @type {number} */
-      let relativeStart;
-      if (start === undefined) {
-        relativeStart = 0;
-      } else {
-        if (start < 0) {
-          relativeStart = Math.max(O.size + start, 0);
-        } else {
-          relativeStart = Math.min(start, O.size);
-        }
-      }
-      /** @type {number} */
-      let relativeEnd;
-      if (end === undefined) {
-        relativeEnd = O.size;
-      } else {
-        if (end < 0) {
```

Comment:
This implements the spec algorithm exactly - is it worth changing? Just makes the diff larger :-)

Reply:
Mostly copied everything from fetch-blob slice method b/c it operates on blobParts `(Uint8Array|Blob)[]` now instead.

could revert/keep this part where relative end/start is calculated